### ISSUE TITLE
Signed Zone offers can't be fulfilled

### ISSUE THREAD

**Issue by mchmatt** [ProjectOpenSea/opensea-js#1134]
### Component

API, Other (please describe)

### Have you ensured that all of these are up to date?

- [X] opensea-js
- [X] Node (minimum v16)

### What version of opensea-js are you on?

6.1.5 (built from main branch)

### What function is the bug in?

fulfillOrder @sdk.ts

### Operating System

Windows

### Describe the bug

Signed Zone offers (read: offers that go through OpenSea's 0x000000e7Ec00e7B300774b00001314B8610022b8 zone) require the extraData field to be sent.

While Seaport-JS supports that field, OpenSea-JS doesn't. Because the extraData field is not sent to the contract, gas estimation (and thus TX sending) fail. Issue https://github.com/ProjectOpenSea/opensea-js/issues/1047 is an example of that, with its error response (0xd232fd2c) decoding to InvalidExtraDataLength(bytes32):
![image](https://github.com/ProjectOpenSea/opensea-js/assets/123990957/6ffab466-f4fe-4069-affb-69f1c2312988)

OS' "fulfill an offer" endpoint provides the required extraData field, so I have [attached a possible solution](https://github.com/ProjectOpenSea/opensea-js/files/12261911/fixed.txt) as an example (don't think that's how the fix should be done)

Is there any reason why OpenSea-JS isn't submitting the transaction as is provided by the API? As in using fulfillment_data.transaction instead of parsing fulfillment_data.orders?

**Comment by mchmatt** [comment]
Simple script to reproduce the issue - make a collection and place an offer (through OS website) on an NFT you own, then try to take that offer through OpenSea-JS
```js
const { ethers } = require("ethers");
const { OpenSeaSDK, Chain } = require("opensea-js");

const RPC_URL = "RPC URL";
const CHAIN = Chain.Sepolia;
const PRIVATE_KEY = "PRIVATE KEY";

//Use an NFT the above wallet owns that has an offer on it
const CONTRACT_ADDRESS = "CONTRACT ADDRESS";
const TOKEN_ID = "TOKEN ID";

const provider = new ethers.providers.JsonRpcProvider(RPC_URL);
const signer = new ethers.Wallet(PRIVATE_KEY, provider);

const openseaSDK = new OpenSeaSDK(
  provider,
  {
    chain: CHAIN,
    apiKey: ""
  },
  undefined,
  signer
);

async function run() {
  const fulfiller = await signer.getAddress();
  const order = await openseaSDK.api.getOrder({
    assetContractAddress: CONTRACT_ADDRESS,
    tokenId: TOKEN_ID,
    side: "bid",
    orderBy: "eth_price",
    orderDirection: "desc"
  })

  await openseaSDK.fulfillOrder({
    order: order,
    accountAddress: fulfiller,
    recipientAddress: fulfiller
  });
}

run();
```

**Comment by ryanio** [comment]
thanks for reporting, we'll look into fixing this

**Comment by mchmatt** [comment]
> @mchmatt I opened a PR to fix (#1141), do you think you could give it a try to see if it works for you? Thanks again.

@ryanio That did fix the issue, appreciate it \:)

**Comment by ryanio** [comment]
have released as [v6.1.7](https://github.com/ProjectOpenSea/opensea-js/releases/tag/v6.1.7)!
